packages/api-explorer/src/scenes/DiffScene/diffUtils.ts

```diff
@@ -118,3 +118,19 @@ export const diffToSpec = (
   result.types = {}
   return result
 }
+
+/**
+ * Returns all valid diff options contained in list
+ * @param opts url diff options parameter value
+ */
+export const getValidDiffOptions = (opts: string | null) => {
+  if (!opts) return null
+  const diffOptions: string[] = []
+  for (const option of opts.split(',')) {
+    const op = option.toLowerCase()
+    if (allDiffToggles.includes(op) && !diffOptions.includes(op)) {
+      diffOptions.push(option.toLowerCase())
+    }
+  }
+  return diffOptions.length ? diffOptions : null
```

Comment:
why not always return an array? i.e why not return an empty array if there are no valid options?

Reply:
I think I faced a weird case earlier when working on this before some of our major changes in main, but yes this makes sense.